lib/scraper.js

```diff
@@ -26,24 +26,49 @@ function loadHtmlAndCss (context, po) {
 
 function Scraper (options) {
 	this.originalResources = [];
+	this.occupiedFileNames = [];
 	this.loadedResources = [];
+	this.respondedResourcePromises = {};
+	this.loadedResourcePromises = {};
 
 	this.options = _.extend({}, defaults, options);
 	this.options.request = _.extend({}, defaults.request, options.request);
 	this.options.directory = path.resolve(process.cwd(), this.options.directory || '');
 	this.options.filenameGenerator = getFilenameGenerator(this.options.filenameGenerator);
 }
 
-Scraper.prototype.getLoadedResource = function getLoadedResource (resource) {
```

Comment:
I meant can we use `respondedResourcePromises` or `loadedResourcePromises` for this purpose?